### ISSUE TITLE
User Defined Chain function not working in reduction metaoperator

### ISSUE THREAD

**Issue by Scimon** [rakudo/rakudo#3541]
# The Problem

From issue #3370 and expanded : 

```
use v6.d;
sub infix:<eog> ( $a,  $b ) is assoc<chain> is pure { 
    so $a == $b+1 
}
say 5 eog 4 eog 3 eog 2 eog 1;
say ( (5 eog 4) and (4 eog 3) and (3 eog 2) and (2 eog 1) );
say [eog] 5,4,3,2,1
```

## Expected Behavior

```
True
True
True
```

## Actual Behavior

```
True
True
False
```

## Environment
* Operating system: Ubuntu Linux 19.10
* Compiler version (`perl6 -v`): Rakudo version 2020.02.1 built on MoarVM version 2020.02.1

Looks like the fix for #3370 handled some chaining but not the metaoperator version.

**Comment by holli-holzer** [comment]
I tried that under windows:
```
D:\>6 simon.p6
False
True
False

D:\>6 -v
This is Rakudo Star version 2019.03.1 built on MoarVM version 2019.03
implementing Perl 6.d.
```
and Ubuntu for WSL
```
holli@DESKTOP-JJASNFC:/mnt/d$ perl6 simon.p6
False
True
False
holli@DESKTOP-JJASNFC:/mnt/d$ perl6 -v
This is Rakudo version 2019.11 built on MoarVM version 2019.11
implementing Perl 6.d.
```

**Comment by Scimon** [comment]
The fix in #3370 was added in the most recent release I believe.

**Comment by vrurg** [comment]
The fix for #3370 was partial. Metaops are apparently unaffected. The output I get on the HEAD is the same as in the report; i.e. the last line is *False*.
